app/controllers/application_controller.rb

```diff
@@ -261,6 +262,15 @@ def user_needs_to_reactivate_account?
     resolved_authn_context_result.identity_proofing?
   end
 
+  def user_recommended_for_piv_cac?
+    current_user.has_gov_or_mil_email? && current_user.piv_cac_recommended_dismissed_at.nil? &&
```

Comment:
Minor: Swapping these could short-circuit to prevent additional DB queries to load email addresses.

```suggestion
    current_user.piv_cac_recommended_dismissed_at.nil? && current_user.has_gov_or_mil_email? &&
```